experimental/InvariantTheory/src/InvariantTheory.jl

```diff
@@ -0,0 +1,523 @@
+    for elemm in new_gens_wrong_ring
+        if elemm != 0
+            push!(new_gens, mixed_to_ring(elemm))
+        end
+    end
+    if length(new_gens) == 0
+        return [ringg()]
+    end
+    
+    #remove ugly coefficients: 
+    V= Vector{FieldElem}[]
+    new_gens_ = Vector{elem_type(ringg)}()
+    for elem in new_gens
+        V = vcat(V,collect(coefficients(elem)))
+        maxx = maximum([abs(denominator(V[i])) for i in 1:length(V)])
+        minn = minimum([abs(numerator(V[i])) for i in 1:length(V)])
+        if denominator((maxx*elem)//minn) != 1
+            error("den not 1")
+        end
+        push!(new_gens_, numerator((maxx*elem)//minn))
+        V= Vector{FieldElem}[]
+    end
+    return new_gens_
+end
+
+#the reynolds operator for SLm acting via new_rep_mat. 
+function reynolds_v_slm(elem::MPolyDecRingElem, new_rep_mat::MatElem, new_det::MPolyDecRingElem, m::Int)
+    mixed_ring_xy = parent(elem)
+    n = ncols(new_rep_mat)
+    new_vars = new_rep_mat*gens(mixed_ring_xy)[1:n]
+    sum_ = mixed_ring_xy()
+    phi = hom(mixed_ring_xy, mixed_ring_xy, vcat(new_vars, [0 for i in 1:ncols(new_rep_mat)+m^2]))
+    sum_ = phi(elem)
+    t = needed_degree(sum_, m)
+    if !is_divisible_by(t, m)
+        return parent(elem)()
+    else
+        p = divexact(t, m)
+    end
+    #num = omegap_(p, new_det, sum_)
+    #den = omegap_(p, new_det, (new_det)^p)
+    #if !(denominator(num//den)==1)
+    #    error("denominator of reynolds not rational")
+    #end
+    #return numerator(num//den)
+    return reynolds_slm(sum_, new_det, p)
+end
+
+#reynolds operator for SLm using Cayleys Omega process 
+function reynolds_slm(elem::MPolyRingElem, det_::MPolyRingElem, p::Int)
+    num = omegap_(p,det_, elem)
+    den = omegap_(p,det_,det_^p)
+    if !(denominator(num//den)==1)
+        error("denominator of reynolds not rational")
+    end
+    return numerator(num//den)
+end
+
+#used to compute the degree p of omega_p
+#computes the degree of the z_ij variables of the leading term of elem.
+function needed_degree(elem_::MPolyDecRingElem, m::Int)
+    elem = leading_monomial(elem_)
+    R = parent(elem)
+    n = ngens(R) - m^2
+    extra_ring, _= polynomial_ring(base_ring(R), "z"=>1:m^2)
+    mapp = hom(R,extra_ring, vcat([1 for i in 1:n], gens(extra_ring)))
+    return total_degree(mapp(elem))
+end
+
+ function omegap_(p::Int, det_::MPolyDecRingElem, f::MPolyDecRingElem)
+     parent(det_) == parent(f) || error("Omega process ring error")
+     action_ring = parent(det_)
+     monos = collect(monomials(det_))
+     coeffs = collect(coefficients(det_))
+     for i in 1:p
+     h = action_ring()
+     for i in 1:length(monos)
+         exp_vect = exponent_vector(monos[i], 1)
+         x = f
+         for i in 1:length(exp_vect)
+             for j in 1:exp_vect[i]
+                 x = derivative(x, i)
+                 if x == 0
+                     break
+                 end
+             end
+         end
```

Comment:
The `break` only aborts the inner loop, the outer loop keeps running. This can be avoided by fusing the loops. Also, check for zero using `is_zero`, not by comparing a polynomial to an `Int` which may require converting the integer into a polynomial.
```suggestion
         for i in 1:length(exp_vect), j in 1:exp_vect[i]
             x = derivative(x, i)
             iszero(x) && break
         end
```